src/database/database-mariadb.cpp

```diff
@@ -0,0 +1,1407 @@
+/*
+ * MariaDB Database Backend for Minetest
+ * Copyright 2023 CaffeinatedBlocks (caffeinatedblocks@gmail.com)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "config.h"
+
+#if USE_MARIADB
+
+#include <cstdlib>
+#include <istream>
+#include <netinet/in.h>
+#include "debug.h"
+#include "exceptions.h"
+#include "settings.h"
+#include "remoteplayer.h"
+#include "server/player_sao.h"
+#include "util/string.h"
+#include "database-mariadb.h"
+
+
+/**
+ * 
+ * 
+ * Database_MariaDB -- the default MariaDB backend implementation
+ * 
+ * 
+ */
+
+
+Database_MariaDB::Database_MariaDB(const std::string &connect_string, const char *type) {
+
+	if (connect_string.empty()) {
+		
+		std::ostringstream oss;
+
+		oss << "[MariaDB] Error: connection string is empty or undefined.\n\n";
+		oss << "Set mariadb" << type << "_connection string in world.mt to use the MariaDB backend.\n\n";
```

Comment:
you can optimize a bit the code by not putting a `;` on each line ending and calling oss only on first line

Reply:
Example: https://linuxhint.com/c-multiline-string-literals/

Reply:
I didn't know the best way to handle multiple lines and throughout the engine code, the style varies, so I went iwth what was done in src/server.cpp at startup with the ASCII art and following lines. In any case, pushing new commit now to resolve.